observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/AkkaHttpMetricsSpec.scala

```diff
@@ -0,0 +1,621 @@
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on failed (exception) response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/exception",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            response.length.toLong
+          )
+        }
+      }
+    }
+
+    "record byte payload size on multiple response" in {
+      withRouteAndMetrics { (route, metrics) =>
+        Get(
+          "/",
+          HttpEntity.Strict(ContentTypes.`text/plain(UTF-8)`, ByteString("hello")),
+        ) ~> route ~> check {
+          responseAs[String] // force processing the response
+        }
+        Get(
+          "/mirror/200",
+          HttpEntity.Chunked(
+            ContentTypes.`application/octet-stream`,
+            Source(List(ByteString(byteArray1), ByteString(byteArray2))),
+          ),
+        ) ~> route ~> check {
+          responseAs[String] // force processing the response
+        }
+        Get(
+          "/exception",
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString(byteArray1)),
+        ) ~> route ~> check {
+          val response = responseAs[String]
+          metrics.httpResponsesBytesTotalValue should be(
+            4L + 15L + response.length.toLong
+          )
+        }
+      }
+    }
+  }
+
+  "request_bytes_total and responses_bytes_total support" should {
+    "not alter string data" in {
+      withRouteAndMetrics { (route, _) =>
+        Get(
+          "/mirror/200",
+          HttpEntity.Strict(ContentTypes.`text/plain(UTF-8)`, ByteString("hello")),
+        ) ~> route ~> check {
+          responseAs[String] should be("hello")
+        }
+      }
+    }
+
+    "not alter binary data" in {
+      withRouteAndMetrics { (route, _) =>
+        Get(
+          "/mirror/200",
+          HttpEntity.Default(
+            ContentTypes.`application/octet-stream`,
+            4,
+            Source.single(ByteString(byteArray1)),
+          ),
+        ) ~> route ~> check {
+          responseAs[Array[Byte]] should equal(byteArray1)
+        }
+      }
+    }
+
+    "record byte payload size on successful response, chuncked" in {
+      withRouteAndMetrics { (route, _) =>
+        Get(
+          "/mirror/200",
+          HttpEntity.Chunked(
+            ContentTypes.`application/octet-stream`,
+            Source(List(ByteString(byteArray1), ByteString(byteArray2))),
+          ),
+        ) ~> route ~> check {
+          responseAs[Array[Byte]] should equal(Bytes.concat(byteArray1, byteArray2))
+        }
+      }
+    }
+
+  }
+
+  "request_duration_seconds" should {
+    "record the time of any request" in {
```

Comment:
To avoid confusion, i.e., time here is not a specific point in time:
```suggestion
    "record duration of any request" in {
```